### PR TITLE
Upgrade zeppelin

### PR DESCRIPTION
**Pull request by finalspy**
Use latest zeppelin image as base
Use latest spark2.4.5 release as well

Use a turnaround because of : https://issues.apache.org/jira/browse/ZEPPELIN-4783
until PR : https://github.com/apache/zeppelin/pull/3755 is merged

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -22,7 +22,7 @@ This container can run without any configuration on Saagie's platform.
 
 If you want to run it with an in-memory Spark, just run:
 ```
-docker run -it --rm --name zeppelin -p 8080:8080 saagie/zeppelin:sp1.12.1-mesos1.3.1-spark2.1.0
+docker run -it --rm --name zeppelin -p 8080:8080 saagie/zeppelin:sp1.12.1-mesos1.3.1-spark2.3.4
```
Comment (tquiviger) [inline README.md]: SPARK_VERSION is 2.4.5 by default, maybe your example could use 2.4.5 as well?

File: README.md [hunk 2]
```diff
@@ -33,7 +33,7 @@ docker run -it --rm --name zeppelin --net=host -e PORT0=[ZEPPELIN_PORT] \
   -v $(pwd)/conf/hadoop/:/etc/hadoop/conf/ \
   -v $(pwd)/conf/spark/spark-env.sh:/usr/local/spark/conf/spark-env.sh \
   -v $(pwd)/conf/spark/spark-defaults.conf:/tmp/spark-defaults.conf \
-  saagie/zeppelin:sp1.12.1-mesos1.3.1-spark2.1.0 /zeppelin/saagie-zeppelin.sh -d DEBUG --port [ZEPPELIN_PORT]
+  saagie/zeppelin:sp1.12.1-mesos1.3.1-spark2.3.4 /zeppelin/saagie-zeppelin.sh -d DEBUG --port [ZEPPELIN_PORT]
```
Comment (tquiviger) [inline README.md]: SPARK_VERSION is 2.4.5 by default, maybe your example could use 2.4.5 as well?

File: Dockerfile [hunk 3]
```diff
@@ -1,62 +1,79 @@
...
+    && rm -rf /var/lib/apt/lists/*
+########################## LIBS PART END ##########################
+
+
+########################## Install Spark BEGIN ##########################
+RUN wget https://archive.apache.org/dist/spark/spark-${SPARK_VERSION}/spark-${SPARK_VERSION}-bin-hadoop${HADOOP_VERSION}.tgz -P /tmp \
+    && tar -zxf /tmp/spark-${SPARK_VERSION}-bin-hadoop${HADOOP_VERSION}.tgz -C /usr/local/  \
+    && ln -s /usr/local/spark-${SPARK_VERSION}-bin-hadoop${HADOOP_VERSION}/ /usr/local/spark \
+    && ln -s /usr/local/spark-${SPARK_VERSION}-bin-hadoop${HADOOP_VERSION}/ /usr/local/spark2 \
+    && rm /tmp/spark-${SPARK_VERSION}-bin-hadoop${HADOOP_VERSION}.tgz
```
Comment (tquiviger) [inline Dockerfile:43]: Is the copy to **/usr/local/spark2** really needed? I remember we did that for Jupyter Spark but can't remember why.

File: saagie-zeppelin.sh [hunk 4]
```diff
@@ -54,14 +54,14 @@ fi
 # before running Zepplin
 if [ -f "/usr/local/spark/conf/spark-env.sh" ]
 then
-  # overwrite Spark 2.1.0 config
+  # overwrite Spark config
   echo "INFO: ovewriting default spark-env.sh"
-  cp /usr/local/spark/conf/spark-env.sh /usr/local/spark/2.1.0/conf
-  chmod 755 /usr/local/spark/2.1.0/conf/spark-env.sh
+  cp /usr/local/spark/conf/spark-env.sh /usr/local/spark/${SPARK_VERSION}/conf
```
Comment (tquiviger) [inline saagie-zeppelin.sh:59]: We're using the SPARK_VERSION to find the correct PATH but Spark 2.4.5 is not installed in **/usr/local/spark/2.4.5** but in **/usr/local/spark-2.4.5-bin-hadoop2.6/**
Consequence => When running any Spark SQL command, for instance `%sql select * from default.test`
: 
_/zeppelin/bin/interpreter.sh: line 254: /usr/local/spark/2.4.5/bin/spark-submit: No such file or directory_

**Review by tquiviger — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (finalspy) [reply README.md]: OK ! This was changed in December when upgrading 2.3.4 but I forgot to change it for this other upgrading.

File: README.md (hunk 2 above)
Comment (finalspy) [reply README.md]: OK ! This was changed in December when upgrading 2.3.4 but I forgot to change it for this other upgrading.

File: Dockerfile (hunk 3 above)
Comment (finalspy) [reply Dockerfile:43]: I too have no idea why it was there but as I'm not supposed to change the image structure I let it like it was before.
If you are sure I can remove it, so I'll.

Comment (finalspy) [reply Dockerfile:43]: Ok this is a typo, in fact I spark2 was supposed to be spark/2.4.5 as necessary ...

File: saagie-zeppelin.sh (hunk 4 above)
Comment (finalspy) [reply saagie-zeppelin.sh:59]: corrected

**Comment by CourbeB** [comment]
Without volume:
Get this error message after running `spark` in a notebook
```
org.apache.zeppelin.interpreter.InterpreterException: java.io.IOException: Error from server (Forbidden): error when creating "STDIN": pods is forbidden: User "system:serviceaccount:vallourec-project474216a0-f500-49d3-9d01-43b5fa80893f:saagie-pods" cannot create pods in the namespace "vallourec-project474216a0-f500-49d3-9d01-43b5fa80893f"
	at org.apache.zeppelin.interpreter.remote.RemoteInterpreter.open(RemoteInterpreter.java:134)
	at org.apache.zeppelin.interpreter.remote.RemoteInterpreter.getFormType(RemoteInterpreter.java:298)
	at org.apache.zeppelin.notebook.Paragraph.jobRun(Paragraph.java:431)
	at org.apache.zeppelin.notebook.Paragraph.jobRun(Paragraph.java:74)
	at org.apache.zeppelin.scheduler.Job.run(Job.java:172)
	at org.apache.zeppelin.scheduler.AbstractScheduler.runJob(AbstractScheduler.java:130)
	at org.apache.zeppelin.scheduler.RemoteScheduler$JobRunner.run(RemoteScheduler.java:159)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.io.IOException: Error from server (Forbidden): error when creating "STDIN": pods is forbidden: User "system:serviceaccount:vallourec-project474216a0-f500-49d3-9d01-43b5fa80893f:saagie-pods" cannot create pods in the namespace "vallourec-project474216a0-f500-49d3-9d01-43b5fa80893f"
	at org.apache.zeppelin.interpreter.launcher.Kubectl.execAndGet(Kubectl.java:141)
	at org.apache.zeppelin.interpreter.launcher.Kubectl.apply(Kubectl.java:56)
	at org.apache.zeppelin.interpreter.launcher.K8sRemoteInterpreterProcess.apply(K8sRemoteInterpreterProcess.java:233)
	at org.apache.zeppelin.interpreter.launcher.K8sRemoteInterpreterProcess.apply(K8sRemoteInterpreterProcess.java:222)
	at org.apache.zeppelin.interpreter.launcher.K8sRemoteInterpreterProcess.start(K8sRemoteInterpreterProcess.java:91)
	at org.apache.zeppelin.interpreter.ManagedInterpreterGroup.getOrCreateInterpreterProcess(ManagedInterpreterGroup.java:65)
	at org.apache.zeppelin.interpreter.remote.RemoteInterpreter.getOrCreateInterpreterProcess(RemoteInterpreter.java:110)
	at org.apache.zeppelin.interpreter.remote.RemoteInterpreter.internal_create(RemoteInterpreter.java:163)
	at org.apache.zeppelin.interpreter.remote.RemoteInterpreter.open(RemoteInterpreter.java:131)
	... 13 more
Caused by: org.apache.commons.exec.ExecuteException: Process exited with an error: 1 (Exit value: 1)
	at org.apache.commons.exec.DefaultExecutor.executeInternal(DefaultExecutor.java:404)
	at org.apache.commons.exec.DefaultExecutor.execute(DefaultExecutor.java:166)
	at org.apache.commons.exec.DefaultExecutor.execute(DefaultExecutor.java:153)
	at org.apache.zeppelin.interpreter.launcher.Kubectl.execute(Kubectl.java:155)
	at org.apache.zeppelin.interpreter.launcher.Kubectl.execAndGet(Kubectl.java:125)
	... 21 more
```
With volume:
Impossible to launch the application on Saagie v2

**Comment by finalspy** [comment]
Everything already merged through #4 (and even more)

**Pull request closed without merging**